barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.hpp

```diff
@@ -138,46 +182,37 @@ std::vector<mul_quad_<fr>> split_into_mul_quad_gates(Acir::Expression const& arg
                                                      std::map<uint32_t, bb::fr>& linear_terms);
 
 /**
- * @brief Given an Acir::Expression and its processed linear terms, determine whether it can be represented by a single
- * width-4 arithmetic gate.
+ * @brief Single entrypoint for handling arithmetic (AssertZero) opcodes.
  *
- * @details By processed linear terms, we mean selector values accumulated per witness index. See process_linear_terms.
- */
-bool is_single_arithmetic_gate(Acir::Expression const& arg, const std::map<uint32_t, bb::fr>& linear_terms);
-
-/**
- * @brief Process the linear terms of an Acir::Expression into a map of witness indices to selector values.
+ * @details This function processes an Acir::Opcode::AssertZero by converting it into one more more mul_quad_ gates. The
+ * function asserts that all the gates produced are non-zero and that the number of gates produced is consistent with
+ * expectation (one gate if the opcode is meant to fit in one gate, more than one gate if the opcode is not meant to fit
+ * in one gate).
  *
- * @details Iterating over the linear terms of the expression, we accumulate selector values for each witness index
  */
-std::map<uint32_t, bb::fr> process_linear_terms(Acir::Expression const& expr);
-
-/**
- * @brief Construct a poly_tuple for a standard width-3 arithmetic gate from its acir representation.
- */
-arithmetic_triple serialize_arithmetic_gate(Acir::Expression const& arg);
-
-/**
- * @brief Assigns a linear term to a specific index in a mul_quad_<bb::fr> gate.
- */
-void assign_linear_term(mul_quad_<bb::fr>& gate, int index, uint32_t witness_index, bb::fr const& scaling);
-
-bool is_assert_equal(mul_quad_<fr> const& mul_quad);
-
 void handle_arithmetic(Acir::Opcode::AssertZero const& arg, AcirFormat& af, size_t opcode_index);
 
 /// ========= MEMORY OPERATIONS ========== ///
 
+/**
+ * @brief Handle memory initialization: create a BlockConstraint storing the entries with which the memory table must be
+ * initialized.
+ *
+ */
 BlockConstraint handle_memory_init(Acir::Opcode::MemoryInit const& mem_init);
 
-bool is_rom(Acir::MemOp const& mem_op);
-
-uint32_t poly_to_witness(const arithmetic_triple poly);
-
+/**
+ * @brief Handle memory operation, either read or write, and update the BlockConstraint type accordingly.
+ *
+ */
 void handle_memory_op(Acir::Opcode::MemoryOp const& mem_op, BlockConstraint& block);
 
 /// ========= BLACKBOX FUNCTIONS ========= ///
 
+/**
+ * @brief Single entrypoint for handling blackbox function opcodes.
+ *
+ */
 void handle_blackbox_func_call(Acir::Opcode::BlackBoxFuncCall const& arg, AcirFormat& af, size_t opcode_index);
```

Comment:
The use of the overlay vague "handle" here (and in the other methods) is another thing that's always bothered me. Would you mind making the comment reflect what this method is doing a little more specifically?